viziphant/array_data.py

```diff
@@ -0,0 +1,163 @@
+import neo
+
+def plot_array(
+        axis, dimension, array_data, direction_data=None, electrode_mask=None, 
+        color_map=None, minc=-np.pi, maxc=np.pi, 
+        arrow_scale_factor=1, summary_arrow_scale_factor=None, rotation=0):
+    """Plots a two-dimensional color plot of array-like data with optional 
+    direction arrows (e.g., gradients) on top.
+
+    Parameters
+    ==========
+    axis : int
+        Handle to axis in which to plot the array.
+    dimension : tuple of int of length two
+        The NxM array dimensions. 
+    array_data : neo.AnalogSignal
+        Contains the signal array of X=NxM signals to plot on the array.
+        Convention: Signal X=0 is the lower left, signal X=M-1 is the lower right,
+        signal X=MxN-1 is the top right. Only a single time point of the array may be
+        given, such that the dimensions of `array_data` are `(1, N*M)`. 
+    direction_data : neo.AnalogSignal
+        Directions to plot as arrow. The function takes complex values in the
+        AnalogSignal that determined direction and magnitude of the arrows.
+        Note: To plot gradients, you need to specify the reverse directions.
+        If
+        set to None, no arrows are plotted. Default: None
+    electrode_mask : numpy.array of bool
+        one-dimensional array of NxM booleans that indicate whether to plot
+        data from each of the NxM electrodes. True at position i will plot the
+        i-th channel in the AnalogSignal object, False will plot a black
+        square. If None, all electrodes are plotted.
+        Default: None
+    color_map : matplotlob.pyplot.cm.get_cmap
+        Colormap to for plotting the data. If None, the current color map is
+        used. 
+        Default: None
+    arrow_scale_factor : float
+        Factor by which to scale the direction arrows in each square.
+        Default: 1
+    summary_arrow_scale_factor : float
+        Factor by which to scale the summary direction arrows in the four array
+        quadrants. If None, these arrows are not plotted.
+        Default: None
+    rotation : float
+        Rotates the array CW by `rotation` degree. Currently partly supported,
+        directions are not yet rotated. Default: 0.
+
+
+    Returns
+    =======
+    numpy.array
+        Image returned by pcolormesh.
+        
+    Example
+    =======
+    # Create fake data for a 4x25 electrode array, that is the array is 4 
+    # electrodes high and 25 wide. Electrode 1 is bottom left, electrode 100 
+    # is top right. We set channel to a high value. Also, for simplicity, we'll
+    # create only one sample of the AnalogSignal.
+    >>> grid_data = np.random.random((1,100))
+    >>> grid_data[0,3] = 3
+    >>> grid_data_obj = neo.AnalogSignal(
+            grid_data*pq.mV, t_start=0*pq.s, sampling_rate=1000*pq.Hz)
+    
+    # Create corresponding directions in the direction 0 rad
+    direction_data = np.exp(np.complex(0,1)*np.random.random((1,100))*.5)
```

Comment:
A fix to a part of the doctest
```suggestion
    >>> direction_data = np.exp(np.complex(0,1)*np.random.random((1,100))*.5)
```